src/main/java/io/confluent/examples/streams/GlobalStoresExample.java

```diff
@@ -0,0 +1,246 @@
+                () -> new GlobalStoreUpdater<>(CUSTOMER_STORE));
+
+        // Create a store table for products. The data from this global store
+        // will be fully replicated on each instance of this application.
+        builder.addGlobalStore(
+                Stores.keyValueStoreBuilder(Stores.persistentKeyValueStore(PRODUCT_STORE), Serdes.Long(), productSerde),
+                PRODUCT_TOPIC,
+                Consumed.with(Serdes.Long(), productSerde),
+                () -> new GlobalStoreUpdater<>(PRODUCT_STORE));
+
+        // We transform each order with a value transformer which will access each
+        // global store to retrieve customer and product linked to the order.
+        final KStream<Long, EnrichedOrder> enrichedOrdersStream = ordersStream.transformValues(() -> new ValueTransformer<Order, EnrichedOrder>() {
+
+            private KeyValueStore<Long, Customer> customerStore;
+            private KeyValueStore<Long, Product> productStore;
+
+            @Override
+            public void init(final ProcessorContext processorContext) {
+                customerStore = (KeyValueStore<Long, Customer>) processorContext.getStateStore(CUSTOMER_STORE);
+                productStore = (KeyValueStore<Long, Product>) processorContext.getStateStore(PRODUCT_STORE);
+            }
+
+            @Override
+            public EnrichedOrder transform(final Order order) {
+                // Get the customer corresponding the order from the customer store
+                final Customer customer = customerStore.get(order.getCustomerId());
+
+                // Get the product corresponding the order from the product store
+                final Product product = productStore.get(order.getProductId());
+
+                return new EnrichedOrder(product, customer, order);
+            }
+
+            @Override
+            public void close() {
+                // No-op
+            }
+        });
+
+        // write the enriched order to the enriched-order topic
+        enrichedOrdersStream.to(ENRICHED_ORDER_TOPIC, Produced.with(Serdes.Long(), enrichedOrdersSerde));
+
+        return new KafkaStreams(builder.build(), streamsConfiguration);
+    }
+
+    // Processor that keeps the global store updated.
+    private static class GlobalStoreUpdater<K, V> implements Processor<K, V> {
+
+        private final String storeName;
+
+        public GlobalStoreUpdater(final String storeName) {
+            this.storeName = storeName;
+        }
+
+        private KeyValueStore<K, V> store;
+
+        @Override
+        public void init(final ProcessorContext processorContext) {
+            store = (KeyValueStore<K, V>) processorContext.getStateStore(storeName);
+        }
+
+        @Override
+        public void process(final K key, final V value) {
+            if (value != null) {
+                store.put(key, value);
+            } else {
+                store.delete(key);
+            }
```

Comment:
Would a single-line `store.put(key, value)` achieve the same thing? If `value` is null, then the `store.put(key, null)` will delete the corresponding key, won't it?

Reply:
Yes, don't know how I did not see this... thank you